proposed/http-message.md

```diff
@@ -393,11 +393,89 @@ Implementations are expected to:
   `Psr\Http\Message\UploadedFileInterface` instance for the given location in
   the tree.
 
+The tree structure referenced should mimic the naming structure in which files
+were submitted. As an example, if a file form element was provided using the name
+`my-form[details][avatar]`, the corresponding tree returned by
+`getUploadedFiles()` should be:
+
+```php
+array(
+    'my-form' => array(
+        'details' => array(
+            'avatar' => /* UploadedFileInterface instance */
+        ),
+    ),
+)
+```
+
+In cases where an array of files are present, the implementation must aggregate
+all information related to the file at the given index. As an example, if
+multiple files were submitted using the file form element named
+`my-form[details][avatars][]`, `$_FILES` would contain the following structure:
+
+```php
+array(
+    'my-form' => array(
+        'details' => array(
+            'avatars' => array(
+                'tmp_name' => array(
+                    0 => '...',
+                    1 => '...',
+                    2 => '...',
+                ),
+                'name' => array(
+                    0 => '...',
+                    1 => '...',
+                    2 => '...',
+                ),
+                'size' => array(
+                    0 => '...',
+                    1 => '...',
+                    2 => '...',
+                ),
+                'type' => array(
+                    0 => '...',
+                    1 => '...',
+                    2 => '...',
+                ),
+                'error' => array(
+                    0 => '...',
+                    1 => '...',
+                    2 => '...',
+                ),
+            ),
+        ),
+    ),
+)
+```
+
+The implementation would convert the above to:
```

Comment:
"The implementation" is rather vague.  You mean a PSR-7 implementation, correct?